client/transactions/list/index.js

```diff
@@ -171,6 +172,18 @@ export const TransactionsList = ( props ) => {
 				i !== all.length - 1 && ', ',
 			] );
 		const riskLevel = <RiskLevel risk={ txn.risk_level } />;
+
+		const customerUrl = addQueryArgs( 'admin.php', {
+			page: 'wc-admin',
+			path: '/customers',
+			search: txn.customer_name,
```

Comment:
I think using `customer_email` would be a better way to filter. It's more unique.

It's also a bit odd that the `Customer Name` column has a link but the `Customer Email` column doesn't. Should both columns link to the filtered Customers list?

Reply:
Actually, I was wondering about that too, but I am afraid to assume customer email is always defined. Can you think of a case where customer name is defined but not email, @DanReyLop?

Looping in @LevinMedia. Should we make customer email a link too?

To show the email column:
<img width="353" alt="Screen Shot 2020-12-22 at 00 52 13" src="https://user-images.githubusercontent.com/73803630/102806820-22345880-43f0-11eb-8028-c22417882018.png">

<img width="608" alt="Screen Shot 2020-12-22 at 00 52 23" src="https://user-images.githubusercontent.com/73803630/102806828-26f90c80-43f0-11eb-9577-7a8b7263e00f.png">

Also, I notice the search autocompletes to `<name> (<email>)` when I typed a name in the search field.

<img width="576" alt="Screen Shot 2020-12-22 at 00 54 39" src="https://user-images.githubusercontent.com/73803630/102807028-848d5900-43f0-11eb-9a8d-f8b9e2cb44da.png">
<img width="392" alt="Screen Shot 2020-12-22 at 00 54 49" src="https://user-images.githubusercontent.com/73803630/102807037-86efb300-43f0-11eb-91b1-d422b7856701.png">

How about:
* If name and email is defined, use <name> (<email>)
* If only email is defined, use <email>
* If only name is defined, use <name>

Reply:
> Can you think of a case where customer name is defined but not email, @DanReyLop?

The code in the `process_payment` flow assumes that the billing email is defined ([source](https://github.com/Automattic/woocommerce-payments/blob/6e9a9ba1010cfd8571a5fef05b04c1ebcf2a060f/includes/class-wc-payment-gateway-wcpay.php#L499-L518)). The merchant could disable the email field in the checkout using filters, but then WCPay would probably break. It's safe to assume we'll always have an email (and if not, not showing a link wouldn't be the end of the world).

> I think customer's email registered to the store can be different than one from billing email, right? At least, that's the case in my local environment.

Interesting... you're right, the billing email doesn't need to be the "account email". I guess there's no perfect solution then. Using the customer name as a "key" (not really a key, just a search filter) seems like a workaround.

Reply:
I did some digging. If you put the Customers page in "Single Customer" mode, you can type the customer name and after you choose one you'll get to a URL like this: `/wp-admin/admin.php?page=wc-admin&path=%2Fcustomers&filter=single_customer&customers=5`, where `5` is a "customer_id" specific for WC-Admin (it's not the "user_id" because there can be guest customers).

I think the ideal solution is to add a "customer_id" field to the transaction in PHP before returning it to JS, using [this function](https://github.com/woocommerce/woocommerce-admin/blob/2964aa9bbe62149261b52c51eef2468f7e1e2a31/src/API/Reports/Customers/DataStore.php#L471) for that (since we already have the Order ID). That way there's no ambiguity, and the Customer name and/or Email can link to that "single customer" view directly.

What do you think? It may be a bit of overkill :)

Reply:
I think that's a good approach. I'll code it up.

By the way, calling `get_existing_customer_id_from_order()` will create N+1 problem, right?
Seems that I need to read what that function actually does and creates a custom query to fetch customer ids given multiple order ids. What do you think?

Reply:
> By the way, calling get_existing_customer_id_from_order() will create N+1 problem, right?
Seems that I need to read what that function actually does and creates a custom query to fetch customer ids given multiple order ids. What do you think?

I wouldn't worry too much about that. WordPress absolutely hammers the database on every request, an extra 25 SELECT queries made [using a PRIMARY KEY](https://github.com/woocommerce/woocommerce-admin/blob/main/src/Install.php#L212) shouldn't make much difference.

Reply:
If you want to optimize the query, you should ask for the best approach to Team Isotope. They may tell you that you can hit the database directly, or they may prefer you open a PR on WC-Admin so the WC-Admin tables are only directly accessed from one place.

Reply:
How about alternative approach:
We could create a wcpay endpoint on the site which would redirect to the required customer page once the link is clicked. We would pass order id and retrieve customer_id from it only when needed rather than on each page request.

Reply:
I like that very much.

Reply:
@shendy-a8c Yes, I think we can link up their email too. Thanks!